src/devices/Tca9548a/Tca9548A.cs

```diff
@@ -0,0 +1,167 @@
+        {
+            return new Tca9548ADevice(this, tcaChannel, address);
+        }
+
+        /// <summary>
+        /// Select a multiplex channel
+        /// </summary>
+        /// <param name="multiplexChannel">The channel to select [2^channel number (0-7)]</param>
+        public void SelectChannel(byte multiplexChannel)
+        {
+            try
+            {
+                _i2CDevice.WriteByte(multiplexChannel);
+                _i2CDevice.ReadByte();
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"Unable to select ({_i2CDevice.ConnectionSettings.BusId} ,{_i2CDevice.ConnectionSettings.DeviceAddress}, {multiplexChannel}): {e}");
+            }
+        }
+
+        /// <summary>
+        /// Select a multiplex channel
+        /// </summary>
+        /// <param name="multiplexChannel">The channel to select</param>
+        public void SelectChannel(Channels multiplexChannel)
+        {
+            SelectChannel(Convert.ToByte(multiplexChannel));
+        }
+
+        /// <summary>
+        /// Try getting the selected channel on Multiplexer
+        /// </summary>
+        /// <param name="selectedChannel"> selected Multiplexer Channel</param>
+        /// <returns> true if able to retrieve selected channel</returns>
+        public bool TryGetSelectedChannel(out Channels selectedChannel)
+        {
+            try
+            {
+                var channel = _i2CDevice.ReadByte();
+                if (Enum.IsDefined(typeof(Channels), channel))
+                {
+                    selectedChannel = (Channels)channel;
+                    return true;
+                }
+                else
+                {
+                    selectedChannel = Channels.Channel0;
+                    return false;
+                }
+            }
+            catch (Exception)
+            {
+                selectedChannel = Channels.Channel0;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Returns all the connected devices on all channels of Multiplexer
+        /// </summary>
+        /// <returns> Dictionary having Keys as Channels and Values as IEnumerable integer type</returns>
+        public Dictionary<Channels, IEnumerable<int>> ScanAllChannelsForDeviceAddress()
+        {
+            Dictionary<Channels, IEnumerable<int>> channels = new Dictionary<Channels, IEnumerable<int>>();
+            foreach (Channels channel in DeviceChannels)
+            {
+                channels.Add(channel, ScanChannelsForDeviceAddress(channel));
+            }
+
+            return channels;
+        }
+
+        /// <summary>
+        /// Returns all the connected device on selected channel
+        /// </summary>
+        /// <param name="channel">Selected Channel on Multiplexer</param>
+        /// <returns></returns>
+        public IEnumerable<int> ScanChannelsForDeviceAddress(Channels channel)
```

Comment:
I'd recommend to replace channel enum with simply `int NumberOfChannels { get; }` and `int` on the arguments. Or alternatively implement `IList<I2cBus>` and then just override `Count` rather than adding `NumberOfChannels`